src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -181,33 +199,23 @@ Pathfinder::Pathfinder (
         mDstAmount (saDstAmount),
         mSrcCurrency (uSrcCurrency),
         mSrcIssuer (uSrcIssuer),
-        mSrcAmount ({uSrcCurrency, uSrcIssuer}, 1u, 0, true),
+        convert_all_ (mDstAmount ==
+            STAmount (mDstAmount.issue (), 1u, 0, true)),
```

Comment:
Here you're making the same comparison again, and this time you're unnecessarily constructing an STAmount (which is a tiny bit expensive because of the hash).

    bool isMinusOne(STAmount const& a) { return a.negative() && a.mantissa() == 1; }

Clearly there's some special case here that I don't fully understand.

Also - why not represent this special case by another `boost::optional`?